### PR TITLE
com_fields : checking for availability of component, correcting 2 fatal errors

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/12645
and for issue https://github.com/joomla/joomla-cms/issues/12633
### Summary of Changes

Prevents fields menu and sidebar from displaying when com_fields is not enabled.
Displays a Warning if the fields system plugin is disabled.
Corrected 2 fatal errors JFolder and JFile not found when saving a custom field.
Alpha ordered and modified the xml_description language string for the system plugin as the custom fields will NOT display if disabled.
### Testing Instructions
Patch
Disable the component com_fields in Extensions Manager=>Manage 
result: menus and sidebar do not show (Users, contacts, content)

Then enable it again and disable the fields system plugin. Display the fields manager
Result : a Warning is displayed in the manager
screen shot 2016-10-31 at 08 03 22

Create a new custom field and save. Before patch fatal error, after, it saves fine

TODO: change the fields manager and fields group manager title to fit the component concerned

### CODE REVIEW

File: administrator/components/com_contact/helpers/contact.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static function addSubmenu($vName)
 			$vName == 'categories'
 		);
 
-		if (JComponentHelper::getComponent('com_contact')->params->get('custom_fields_enable', '1'))
+		if (JFields::isEnabled() && JComponentHelper::getComponent('com_contact')->params->get('custom_fields_enable', '1'))
```
Comment (andrepereiradasilva) [inline administrator/components/com_contact/helpers/contact.php]: Can't we use just JComponentHelper::isEnabled('com_fields') ?

Comment (infograf768) [inline administrator/components/com_contact/helpers/contact.php]: the new jfields ii created takes care of both the component and the plugin, but, yes, we could separate these and use the new jfields to take only care of the plugin, in order to use it only for the message in the manager. will look tomorrow.

Comment (infograf768) [inline administrator/components/com_contact/helpers/contact.php]: and same for the plugin

**Comment by infograf768** [comment]
Modified test instructions.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12646">issues.joomla.org/tracker/joomla-cms/12646</a>.</sub>

**Comment by infograf768** [comment]
@laoneo
@brianteeman
@andrepereiradasilva 
@coolcat-creations
@rdeutz 

Please test. That is one of the multiple PRs to correct com_fields.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on c05c148ed6490348eff50d90292bf023621fa3cc<br /><br />Works as described.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12646">issues.joomla.org/tracker/joomla-cms/12646</a>.</sub>

**Comment by wilsonge** [comment]
As @stevejburge tested this successfully (mentioned in #11833) merging this with two good tests

**Pull request merged**